### PR TITLE
Add purchase retention days

### PR DESCRIPTION
**Pull request by bobbyiliev**
If we want to be able to run the datagen script 24/7 we need a way to clear the old purchases data otherwise it will fill up the RDS disk space very quickly.

Changes:
- Define an optional `toInfinity` loop that will let you run the script 24/7 rather than just 500k loops. If `toInfinity` is set to false, the script will loop `purchaseGenCount` times
- Define a `purchaseRetentionDays` which delete any orders older than that value

### CODE REVIEW

**Review by joacoc — APPROVED** [review]
Looks good to me! The only thing calling my attention is the `DELETE` present in every loop. It is just a performance thing. Nothing critical.

**Comment by bobbyiliev** [comment]
> Looks good to me! The only thing calling my attention is the DELETE present in every loop. It is just a performance thing. Nothing critical.

Good point @joacoc! An alternative option that comes to mind is maybe having a separate script, that we run as a corn job every hour or so.

File: ecommerce/loadgen/generate_load.py [hunk 1]
```diff
@@ -138,6 +149,10 @@ def generatePageview(viewer_id, target_id, page_type):
                 )
                 connection.commit()
 
+                # Delete old purchases (older than purchaseRetentionDays days)
+                cursor.execute(
```
Comment (dehume) [inline ecommerce/loadgen/generate_load.py:153]: Do you have to `commit` this execution?

**Review by dehume — APPROVED** [review]
~~Looks good though a little worried about putting the delete in the loop, though not seeing anywhere else it can go for this script~~

Joaquin beat me to it

**Comment by morsapaes** [comment]
> An alternative option that comes to mind is maybe having a separate script, that we run as a cron job every hour or so.

I'm +1 on separating concerns here so the whole thing doesn't become a nightmare to maintain.

**Review by morsapaes — APPROVED** [review]

**Pull request merged**